src/Func.h

```diff
@@ -1990,13 +2003,101 @@ class Func {
      */
     // @{
     Func &prefetch(const Func &f, const VarOrRVar &var, Expr offset = 1,
-                   PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf);
+                   PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf) {
+        return prefetch_at(f, var, var, std::move(offset), strategy);
+    }
     Func &prefetch(const Internal::Parameter &param, const VarOrRVar &var, Expr offset = 1,
-                   PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf);
+                   PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf) {
+        return prefetch_at(param, var, var, std::move(offset), strategy);
+    }
     template<typename T>
     Func &prefetch(const T &image, VarOrRVar var, Expr offset = 1,
                    PrefetchBoundStrategy strategy = PrefetchBoundStrategy::GuardWithIf) {
-        return prefetch(image.parameter(), var, offset, strategy);
+        return prefetch_at<T>(image, var, var, std::move(offset), strategy);
+    }
+    // @}
+
+    /** prefetch_at() is a more fine-grained version of prefetch(), which allows
+     * specification of different vars for the location of the prefetch() instruction
+     * vs. the location that is being prefetched:
+     *
+     * - the first var specified, 'loop', indicates the loop in which the prefetch will be placed
```

Comment:
I think there must be better names for these, at least `loop`. Both options are loops... I think the first var should be something related to `at`, given the similarity to `compute_at` and `store_at`. `prefetch_at`, and `prefetch_from` for the other variable? Or maybe just `at` and `from` given the context.

Reply:
Used `at` and `from`, PTAL